File: database/models/task_progress.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey, Index
from database.base import Base


class TaskProgress(Base):
    __tablename__ = "task_progress"

    id = Column(Integer, primary_key=True, autoincrement=True)
    task_id = Column(String(64), ForeignKey("task_entry.id", ondelete="CASCADE"), nullable=False)
    seq = Column(Integer, nullable=False)
    message = Column(Text, nullable=False)

    __table_args__ = (Index("ix_task_progress_task_id_seq", "task_id", "seq"),)

File: zihuan_core/src/database/ddl/task_progress.rs
pub const MYSQL_DDL: &str = "CREATE TABLE IF NOT EXISTS task_progress (
        id INTEGER PRIMARY KEY AUTO_INCREMENT,
        task_id VARCHAR(64) NOT NULL,
        seq INTEGER NOT NULL,
        message TEXT NOT NULL,
        CONSTRAINT fk_task_progress_task
            FOREIGN KEY (task_id) REFERENCES task_entry(id)
            ON DELETE CASCADE
    ) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4";

pub const SQLITE_DDL: &str = "CREATE TABLE IF NOT EXISTS task_progress (
        id INTEGER PRIMARY KEY AUTOINCREMENT,
        task_id TEXT NOT NULL,
        seq INTEGER NOT NULL,
        message TEXT NOT NULL,
        FOREIGN KEY (task_id) REFERENCES task_entry(id) ON DELETE CASCADE
    )";

pub const MYSQL_INDEXES: &[&str] = &[
    "CREATE INDEX IF NOT EXISTS ix_task_progress_task_id_seq ON task_progress (task_id, seq)",
];

pub const SQLITE_INDEXES: &[&str] = &[
    "CREATE INDEX IF NOT EXISTS ix_task_progress_task_id_seq ON task_progress (task_id, seq)",
];
